session.go

```diff
@@ -490,24 +493,36 @@ runLoop:
 
 		s.maybeResetTimer()
 
+		var sendQueueChan <-chan struct{}
```

Comment:
Maybe add a comment on why we don't always wait on this chan, even if we're not blocked? I think it's because of spurious unblocks, but I'm not sure.

Reply:
Done.